framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -337,7 +337,43 @@ IterationAdaptiveDT::limitDTByFunction(Real & limitedDT)
 {
   Real orig_dt = limitedDT;
 
-  if (_timestep_limiting_function)
+  // Limit by function change for piecewise linear functions.
+  if (_piecewise_timestep_limiting_function && _max_function_change > 0)
+  {
+    auto linear_function = dynamic_cast<PiecewiseLinear *>(_piecewise_timestep_limiting_function);
```

Comment:
You might consider saving this cast off as a member so you don't have to perform it each time step.

Reply:
done